src/server/game/AI/ScriptedAI/ScriptedCreature.cpp

```diff
@@ -547,6 +547,23 @@ void BossAI::UpdateAI(uint32 diff)
     DoMeleeAttackIfReady();
 }
 
+void BossAI::_DespawnAtEvade(uint32 delayToRespawn)
+{
+    uint32 corpseDelay = me->GetCorpseDelay();
+    uint32 respawnDelay = me->GetRespawnDelay();
+
+    me->SetCorpseDelay(1);
+    me->SetRespawnDelay(delayToRespawn);
```

Comment:
ASSERT is used for illogic cases that are never supposed to happen, passing the number 2 here  doesn't apply to that case. If you really want to notify the developer use TC_LOG_ERROR and std:max(delayToRespawn, 2) - 1